irods/test/test_ssl_context.bats

```diff
@@ -0,0 +1,145 @@
+# not the iRODS service account) iinit'ed as rods in a default iRODS
+# server installation.
+
+# iRODS RELATED SETUP
+#
+IRODS_LOCAL_ENV=~/.irods/irods_environment.json
+IRODS_ACCOUNT_ENV=~irods/.irods/irods_environment.json
+
+edit_core_re () {
+  if [ "$1" = ssl ]; then
+    sudo su irods -c "sed -i.orig 's/\(^\s*acPreConnect.*CS_NEG\)\([A-Z_]*\)/\1_REQUIRE/' /etc/irods/core.re"
+  else
+    sudo su irods -c "sed -i.ssl 's/\(^\s*acPreConnect.*CS_NEG\)\([A-Z_]*\)/\1_DONT_CARE/' /etc/irods/core.re"
+  fi
+}
+
+restart_server()
+{
+  sudo su irods -c '~/irodsctl restart'
+}
+
+if [ "$LOGFILE" = "<syslog>" ]; then
+  log () { logger "`date`: $*"; } 		# Log to (r)syslog
+elif [ -n "$LOGFILE" ]; then
+  log () { echo "`date`: $*" >>"$LOGFILE" ; }	# Log to a file.
+else
+  log () { :; }			 		# NOP
+endif
+
+: ${REPO:=~/python-irodsclient}
+REPO_SCRIPTS="$REPO/irods/test"
+PATH=$REPO_SCRIPTS:$PATH
+
+touch /tmp/run
+if [ "`cat /tmp/run`" != "$RUN" ]; then
+  ## -- Begin one-time initialization --
+  #
+  json_config -i $IRODS_LOCAL_ENV --clear-store \
+    VAR='irods_ssl_*ca_certificate_file'\
+    VAR='irods_ssl_*ca_certificate_path'\
+    VAR='irods_ssl_*verify_server'\
+    VAR='irods_ssl_*dh_params_file'\
+    VAR='irods_ssl_*certificate_key_file'\
+    VAR='irods_ssl_*certificate_chain_file'\
+    VAR='irods_*encryption_algorithm'\
+    VAR='irods_*encryption_key_size'\
+    VAR='irods_*encryption_num_hash_rounds'\
+    VAR='irods_*encryption_salt_size'\
+    VAR='irods_*client_server_policy'\
+    VAR='irods_*client_server_negotiation'
+
+  # -- Set up the basic server cert
+  [ -e /etc/irods/ssl ] || sudo su irods -c "$REPO_SCRIPTS/setupssl.py -f"
+  # -- Set up another cert with non-matching hostname
+  sudo su irods -c "$REPO_SCRIPTS/setupssl.py -kf -x.localhost -hlocalhost"
+  sudo su irods -c "c_rehash /etc/irods/ssl"
+
+  sudo $REPO_SCRIPTS/json_config -i $IRODS_ACCOUNT_ENV\
+    'client_server_policy="CS_NEG_REQUIRE"'\
+    'ca_certificate_file="/etc/irods/ssl/irods.crt"'\
+    'certificate_key_file="/etc/irods/ssl/irods.key"'\
+    'dh_params_file="/etc/irods/ssl/dhparams.pem"'\
+    'certificate_chain_file="/etc/irods/ssl/irods.crt"'\
+    'verify_server="cert"'
+
+  json_config -i $IRODS_LOCAL_ENV\
+    'client_server_negotiation="request_server_negotiation"'\
+    'encryption_algorithm="AES-256-CBC"'\
+    'encryption_key_size=32'\
+    'encryption_num_hash_rounds=16'\
+    'encryption_salt_size=8'\
+    'client_server_policy="CS_NEG_REQUIRE"'\
+    'verify_server="cert"'\
+    'ca_certificate_file="/etc/irods/ssl/irods.crt"'
+
+  sudo $REPO_SCRIPTS/json_config -i $IRODS_ACCOUNT_ENV -i $IRODS_LOCAL_ENV PRESERVE
+
+  restart_server
+
+  edit_core_re ssl
+
+  trap 'log "Tests Finalizing..."
+        sudo $REPO_SCRIPTS/json_config -i $IRODS_ACCOUNT_ENV -i $IRODS_LOCAL_ENV RESTORE
+        edit_core_re RESTORE
+  ' exit
+  #
+  ## --  End one-time init  --
+  echo "$RUN" >/tmp/run
+fi
+
+# TEST-RELATED
+
```

Comment:
document further, in comments and maybe in README. Assumptions for how the tests and initial conditions (setup/teardown) work and what they do, etc.

Reply:
The BATS tests have been documented  in comments, in the test script itself.